src/box_utils.cpp

```diff
@@ -182,4 +182,90 @@ Box padBox(const Box& box, double padding)
 
   return box_padded;
 }
+
+std::vector<std::vector<std::vector<float>>> parseVVVF(const std::string& input, std::string& error_return)
+{
+  std::stringstream input_ss(input);
+  int depth = 0;
+
+  std::vector<float> current_vector;
+  std::vector<std::vector<float>> current_vvf;
+  std::vector<std::vector<std::vector<float>>> result;
+
+  while (!!input_ss && !input_ss.eof())
```

Comment:
I didn't know how to use `!!`.
The purpose is to convert the `input_ss` into bool and reverse it?
Is this normal?

Reply:
Thank you. This code is a copy of polygon_utils written by other developers. I will check it.
https://github.com/smilerobotics/laser_filters/blob/5777e4ee66ca1078e037591cdb24ad7e9cc4192b/src/polygon_utils.cpp#L88

Reply:
I think it is not a wrong representation.

 `!` operator returns `ture` if some errors occur in the stringstream, and the additional `!` reverses it as you guess. As a result, `!!` returns `true` if there exist no error.
https://en.cppreference.com/w/cpp/io/basic_ios/operator!